app/src/main/java/fr/free/nrw/commons/nearby/NearbyListFragment.java

```diff
@@ -32,80 +26,103 @@
 
 import fr.free.nrw.commons.R;
 
-public class NearbyListFragment extends ListFragment {
+public class NearbyListFragment extends ListFragment implements TaskListener {
 
-    private int mImageSize;
-    private boolean mItemClicked;
+    private NearbyAsyncTask nearbyAsyncTask;
     private NearbyAdapter mAdapter;
+    private ListView listview;
+
+    private ProgressBar progressBar;
```

Comment:
We are already using a progress bar... it's just an indeterminate one, the circle that goes round and round. :) Switching to one that shows % of progress would be nice though, but I guess not high priority?